lib/ansible/plugins/httpapi/nxos.py

```diff
@@ -27,13 +28,19 @@ def _run_queue(self, queue, output):
         request = request_builder(queue, output)
         headers = {'Content-Type': 'application/json'}
 
-        response, response_text = self.connection.send('/ins', request, headers=headers, method='POST')
         try:
-            response_text = json.loads(response_text)
+            response, response_data = self.connection.send('/ins', request, headers=headers, method='POST')
+        except HTTPError:
+            raise AnsibleConnectionFailure('Could not connect to {0}: {1}'.format(
+                self.connection._url + '/ins', exc.reason
+            ))
+
+        try:
+            response_data = json.loads(response_data.getvalue())
```

Comment:
This might fail on python 3 and json.loads expects string and getvalue() will return bytes .

Reply:
On 3.5, yeah... I'll fix that